common/util.go

```diff
@@ -407,6 +408,24 @@ func ValidateCronSchedule(cronSchedule string) error {
 	return nil
 }
 
+// GetBackoffForNextCronSchedule calculates the backoff time for the next run given
+// a cronSchedule and current time
+func GetBackoffForNextCronSchedule(cronSchedule string, nowTime time.Time) time.Duration {
+	if len(cronSchedule) == 0 {
+		return NoRetryBackoff
+	}
+
+	schedule, err := cron.ParseStandard(cronSchedule)
+	if err != nil {
+		return NoRetryBackoff
```

Comment:
It seems weird to me that we don't surface the error if cron schedule is invalid. What about changing this method's API to return (time.Duration, error). The only real error case I see is cron schedule is not valid, all other cases we can return nil for error.